pkgs/applications/misc/1password-gui/default.nix

```diff
@@ -1,61 +1,92 @@
-{ lib
-, stdenv
-, fetchurl
+{ fetchurl
 , makeWrapper
-, electron_11
-, openssl
+, alsaLib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, cairo
+, cups
+, dbus
+, expat
+, gdk-pixbuf
+, glib
+, gtk3
+, libdrm
+, libxkbcommon
+, mesa
+, nspr
+, nss
+, pango
+, systemd
+, libX11
+, libxcb
+, libXcomposite
+, libXdamage
+, libXext
+, libXfixes
+, libXrandr
+, libxshmfence
+, lib
+, xdg-utils
+, stdenv
 }:
-
 stdenv.mkDerivation rec {
   pname = "1password";
-  version = "8.0.33-53.BETA";
+  version = "8.0.34";
 
   src = fetchurl {
-    url = "https://downloads.1password.com/linux/tar/beta/x86_64/1password-${version}.x64.tar.gz";
-    hash = "sha256-YUYER+UiM1QEDgGl0P9bIT65YVacUnuGtQVkV91teEU=";
+    url = "https://downloads.1password.com/linux/tar/stable/x86_64/1password-${version}.x64.tar.gz";
+    sha256 = "0mp119v5vgsva7pnxpsbq4xhh4vbhwv7ga9b5b7f6slx3biy1wmh";
   };
 
   nativeBuildInputs = [ makeWrapper ];
 
   dontConfigure = true;
   dontBuild = true;
+  dontPatchELF = true;
 
-  installPhase = let
-    runtimeLibs = [
-      openssl.out
-      stdenv.cc.cc
-    ];
-  in ''
-    mkdir -p $out/bin $out/share/1password
-
-    # Applications files.
-    cp -a {locales,resources} $out/share/${pname}
-    install -Dm0755 -t $out/share/${pname} {1Password-BrowserSupport,1Password-KeyringHelper}
+  rpath = lib.makeLibraryPath [
+    alsaLib
+    at-spi2-atk
+    at-spi2-core
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    gdk-pixbuf
+    glib
+    gtk3
+    libdrm
+    libxkbcommon
+    mesa
+    nspr
+    nss
+    pango
+    systemd
+    libX11
+    libxcb
+    libXcomposite
+    libXdamage
+    libXext
+    libXfixes
+    libXrandr
+    libxshmfence
+  ] + ":${stdenv.cc.cc.lib}/lib64";
 
-    # Desktop file.
-    install -Dt $out/share/applications resources/${pname}.desktop
-    substituteInPlace $out/share/applications/${pname}.desktop \
-      --replace 'Exec=/opt/1Password/${pname}' 'Exec=${pname}'
-
-    # Icons.
-    cp -a resources/icons $out/share
-
-    # Wrap the application with Electron.
-    makeWrapper "${electron_11}/bin/electron" "$out/bin/${pname}" \
-      --add-flags "$out/share/${pname}/resources/app.asar" \
-      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeLibs}"
+  installPhase = ''
+    mkdir -p $out/bin $out/share/1password
```

Comment:
Please add `runHook preInstall` to the top, and `runHook postInstall` to the bottom of the `installPhase`.

As it is now, if anybody ever overrides the package and tries to do something in `preInstall`, nothing will happen. Once you add the `runHook`s, then this supposed override will work.

Reply:
Since I'm about to have more than one commit, should I squash them and apply that title to the one resulting commit?

Reply:
i didn't at all know about runHook commands! Thanks!

Reply:
Yes, please!